src/rstar.jl

```diff
@@ -88,11 +88,11 @@ function rstar(
     # train classifier on training data
     ycategorical = MLJModelInterface.categorical(y)
     fitresult, _ = MLJModelInterface.fit(
-        classifier, verbosity, Tables.table(x[train_ids, :]), ycategorical[train_ids]
+        classifier, verbosity, Tables.table(x[:, train_ids]'), ycategorical[train_ids]
```

Comment:
A bit unfortunate that the memory layout does not match the MLJ conventions anymore in this case. But probably unavoidable? Or should the new convention only apply to 3d arrays?

Reply:
It might be confusing to the user if the order of two dimensions was swapped when comparing the 2 methods.

Also, even though MLJ assumes columns are features, not every classifier does, so sometimes internally the table is transposed. (see https://alan-turing-institute.github.io/MLJ.jl/dev/quick_start_guide_to_adding_models/#Regressor)

I wonder if it makes sense to relax the constraint that `X` be a matrix input to this function and instead require just that it be table whose columns are parameters. This only makes sense if we can use MLJBase API functions to do the subsetting within the method. Then the 3d-array method would just construct the necessary Table.

Reply:
e.g. it looks like `partition` would work: https://juliaai.github.io/MLJBase.jl/stable/datasets/#MLJBase.partition-Tuple{Any,%20Vararg{Real}}

Reply:
We don't use MLJBase but only MLJModelInterface since that would pull in additional dependencies. There are some design issues in MLJ though it seems since MLJModelInterface is not completely consistent (see, e.g., the comment in the file and the internal workaround I had to add recently).

Reply:
But in principle, I like the idea of supporting generic tables. In that case it might be natural to even just make the chain index another column?

Reply:
I was able to support generic tables, `AbstractMatrix`, and `AbstractVector` inputs using just MLJModelInterface utilities.

> In that case it might be natural to even just make the chain index another column?

Potentially, but there are 2 reasons not to do this. First, whether constructing the inputs from a vector of matrices of draws from different chains or constructing from a 3d array, it's easy to make a table of predictors, but it's another step for the user to append the column of chain IDs to this. Second, MLJ's interface separates the two, and since one of the arguments is constrained to be an MLJ type, it makes sense to be consistent with MLJ's interface.